script/BuyNRT.s.sol

```diff
@@ -1,123 +1,61 @@
-// SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: MIT
 pragma solidity =0.8.17;
 
 import "forge-std/Script.sol";
-import "../contracts/BridgedPolygonNORI.sol";
-import "../contracts/Market.sol";
+import "@/contracts/IERC20WithPermit.sol";
+import "@/test/helpers/signature-utils.sol";
+import "@/contracts/Market.sol";
 
 /**
- * Example script demonstrating a purchase interaction with Nori's Market contract.
+ * @notice Example script demonstrating a purchase interaction with Nori's Market contract.
  *
- * Contextual information is bound in using environment variables, i.e. for Mumbai
+ * Contextual information is bound in using environment variables, i.e. for Mumbai:
  *
- * export MARKET_ADDRESS="0x61A9d9A34Dbc3a1accD55D684A2bF0e0D394201f"
- * export BP_NORI_ADDRESS="0xB3fe45C08137dD6adACb2918D899e0C0dBB036C8"
- * export BUYER_PRIVATE_KEY="<0x00>"
+ * export PURCHASING_TOKEN_ADDRESS="0x0FA8781a83E46826621b3BC094Ea2A0212e71B23" # USDC
+ * export TONNES=1
+ * export MNEMONIC="..."
  * export RECIPIENT_ADDRESS="<0x00>"
  * export MUMBAI_RPC_URL="https://..."
+ * export NETWORK_NAME="mumbai"
  *
- * forge script script/BuyNRT.s.sol:BuyNRT --rpc-url "$MUMBAI_RPC_URL" --private-keys "$BUYER_PRIVATE_KEY" --broadcast -vvvv
+ * forge script script/BuyNRT.s.sol:BuyNRT --rpc-url="$NETWORK_NAME" --mnemonics="$MNEMONIC" --broadcast -vvvv
  */
-
-struct Permit {
-  address owner;
-  address spender;
-  uint256 value;
-  uint256 nonce;
-  uint256 deadline;
-}
-
 contract BuyNRT is Script {
-  /**
-   * @dev This hash is the result of
-   * `keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)")`.
-   */
-  bytes32 public constant PERMIT_TYPEHASH =
-    0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
-
-  BridgedPolygonNORI private _bridgedPolygonNORI;
-  Market private _market;
-
-  function run() external {
-    uint256 buyerPrivateKey = vm.envUint("BUYER_PRIVATE_KEY");
-    address buyerAddress = vm.addr(buyerPrivateKey);
+  SignatureUtils private immutable _signatureUtils = new SignatureUtils();
```

Comment:
fwiw I was spelling out the permit implementation explicitly because I was trying to make this script as free of dependencies as possible since it serves as a simple implementation for partners who might want to integrate.  I think this is probably still a good cleanup, they can always pull in that helper code.